src/utils.jl

```diff
@@ -589,6 +589,15 @@ respectively) can be evaluated using the syntax `var"%3"` and `var"@_4"` respect
 """
 function eval_code end
 
+function extract_usage!(s::Set{Symbol}, expr::Expr)
+    for arg in expr.args
+        extract_usage!(s, arg)
+    end
+    return s
+end
+extract_usage!(s::Set{Symbol}, sym::Symbol) = push!(s, sym)
+extract_usage!(s::Set{Symbol}, ::Any) = s
```

Comment:
This should avoid some dynamic dispatch:

```suggestion
function extract_usage!(s::Set{Symbol}, expr::Expr)
    for arg in expr.args
        if arg isa Symbol
            push!(s, arg)
        elseif arg isa Expr
            extract_usage!(s, arg)
        end
    end
    return s
end
```